### ISSUE TITLE
Remove ddr-densho-122 MPEG2/OGV from Internet Archive

### ISSUE THREAD

**Issue by gjost** [denshoproject/ddr-cmdln#193]
@GeoffFroh 
We have an agreement with Frank Abe that the high-quality video materials from ddr-densho-122 were to be made streamable, but not downloadable from the DDR. To support this, our tech contact at IA (Jake) placed the objects in a special collection there called "stream_only" which is supposed to prevent the originals from being accessible as they would in the case of a regular collection (see, e.g., https://archive.org/details/ddr-densho-122-23-1 vs. https://archive.org/details/ddr-densho-1000-1-1). 

From the IA website itself, the user is prevented from either downloading individual files or seeing the file xml manifest. We've also confirmed that Windows 10 and Ubuntu desktop users receive an error message from IA when attempting to use the download buttons on the DDR site. The strange thing is that MacOS users *are* able to access the content, and right-click "Save As..." apparently works from Windows (with Chrome, Firefox and MS Edge). 

As temporary remediation, we've removed the download buttons from the DDR pages (see https://github.com/denshoproject/ddr-public/issues/165). This won't prevent people who guess the urls from downloading the content, but will definitely deter casual users from accessing it. The more permanent fix involves removing the mezzanine MPEG2's (and OGV derivatives) from IA altogether.

@sarabeckman would you please prep a list of all the ddr-densho-122 entities that are VHs, along with segment counts? If you pass this along to @gjost , he will have an inventory of the ddr IDs to target.

@gjost  I started mucking around with the idea of using the IA python library to do this. It looks like the delete method (https://archive.org/services/docs/api/internetarchive/api.html#internetarchive.delete) has a `formats` arg that you can pass along with the IA identifier (which is fortunately the same as our DDR ID) that allow you to specify which files to remove. This way we don't have to know the actual filename in advance. I haven't tested any of this, however...

I did create a test object on IA that you can use. It doesn't have an OGG file but does have the original MPEG2. The object is at:
https://archive.org/details/TESTddr-densho-delete

For reference, the files meta is here:
https://ia801404.us.archive.org/25/items/TESTddr-densho-delete/TESTddr-densho-delete_files.xml

The second part of the more permanent fix is to adjust the templates so that they don't 500 if the expected MPEG2 url data is not present. We've already talked about this a little, and there are two issues in github that give the details:
https://github.com/denshoproject/ddr-public/issues/163
https://github.com/denshoproject/ddr-cmdln/issues/191

Now that the quick-n-dirty fix is in place, we're not quite as rushed in getting this done, but do want to get the content removed in the near future.

**Comment by gjost** [comment]
@sarabeckman 
Here are all the VH entities in ddr-densho-122 with segment counts:

ddr-densho-122-4 (22 segments)
ddr-densho-122-5 (10 segments)
ddr-densho-122-6 (16 segments)
ddr-densho-122-7 (3 segments)
ddr-densho-122-8 (9 segments)
ddr-densho-122-9 (10 segments)
ddr-densho-122-10 (12 segments)
ddr-densho-122-11 (10 segments)
ddr-densho-122-12 (15 segments)
ddr-densho-122-13 (11 segments)
ddr-densho-122-14 (7 segments)
ddr-densho-122-15 (10 segments)
ddr-densho-122-16 (2 segments)
ddr-densho-122-17 (9 segments)
ddr-densho-122-18 (10 segments)
ddr-densho-122-19 (19 segments)
ddr-densho-122-20 (23 segments)
ddr-densho-122-21 (19 segments)
ddr-densho-122-22 (18 segments)
ddr-densho-122-23 (29 segments)
ddr-densho-122-24 (6 segments)
ddr-densho-122-25 (8 segments)
ddr-densho-122-26 (7 segments)
ddr-densho-122-27 (2 segments)
ddr-densho-122-28 (2 segments)
ddr-densho-122-29 (10 segments)

**Comment by gjost** [comment]
Wrote a script to delete the MPEG2/Ogg Video files, ran it, and it looks like it worked.
```
import internetarchive

ACCESS_KEY = 'REDACTED'
SECRET_KEY = 'REDACTED'

def delete_mpeg2_ogv(eid, access_key=None, secret_key=None):
    print(eid)
    entity = internetarchive.get_item(eid)
    segments = [s for s in entity.contents()]
    print(segments)
    num_segments = len(segments)
    for n,segment in enumerate(entity.contents()):
        print(n,num_segments,segment)
        mpeg2_ogv = [
            f for f in segment.files
            if ('MPEG2' in f['format']) or ('Ogg Video' in f['format'])
        ]
        num_files = len(mpeg2_ogv)
        for n,data in enumerate(mpeg2_ogv):
            f = internetarchive.File(segment, data['name'])
            print(n,num_files,f)
            result = f.delete(access_key=access_key, secret_key=secret_key)
            print(result)

entities = [
    'ddr-densho-122-4', 'ddr-densho-122-5', 'ddr-densho-122-6', 'ddr-densho-122-7',
    'ddr-densho-122-8', 'ddr-densho-122-9', 'ddr-densho-122-10', 'ddr-densho-122-11',
    'ddr-densho-122-12',
    'ddr-densho-122-13', 'ddr-densho-122-14', 'ddr-densho-122-15', 'ddr-densho-122-16',
    'ddr-densho-122-17', 'ddr-densho-122-18', 'ddr-densho-122-19', 'ddr-densho-122-20',
    'ddr-densho-122-21',
    'ddr-densho-122-22', 'ddr-densho-122-23', 'ddr-densho-122-24', 'ddr-densho-122-25',
    'ddr-densho-122-26', 'ddr-densho-122-27', 'ddr-densho-122-28', 'ddr-densho-122-29',
]
for eid in entities:
    delete_mpeg2_ogv(eid, access_key=ACCESS_KEY, secret_key=SECRET_KEY)

```

**Comment by gjost** [comment]
Fixed in `ddr-cmdln` commit `72e0c32` and merged into `develop branch.
